src/actions/connectionKeyPairActions.js

```diff
@@ -257,6 +257,18 @@ export const updateConnectionKeyPairs = (
     const { currentConnectionsCount, oldConnectionsCount, newReceivedConnectonsCount } = numberOfConnections;
     const totalConnections = currentConnectionsCount + oldConnectionsCount + newReceivedConnectonsCount;
 
+    if (generateKeys && totalConnections === 0 && connectionKeyPairs.length === 0 && lastConnectionKeyIndex === -1) {
```

Comment:
Pls put a description under the `// NOTE:` here why do we need this

Reply:
92969f7